File: Examples/SampleAppHyperLinkCustomAction/HyperLinkExample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace HyperLinkExample
{
    public partial class Form1 : Form
    {
        private string file = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory,
            @"..\", @"..\", @"..\", @"..\", @"..\",  @"SqliteExamples\SimpleTest1.rdl");

        public Form1()
        {
            InitializeComponent();
        }

        private async void ButtonReloadReport_Click(object sender, EventArgs e)
        {
           await LoadReport();
        }

        private async Task LoadReport()
        {
            await rdlViewer1.SetSourceFile(new Uri(file));
            await rdlViewer1.Rebuild();
        }

        private void rdlViewer1_Hyperlink(object source, Majorsilence.Reporting.RdlViewer.HyperlinkEventArgs e)
        {
            // report LastName field as HyperLink action
            var url = new Uri(e.Hyperlink);
            if (url.Scheme == "lastname" )
            {
                e.Cancel = true;
                MessageBox.Show(url.ToString());
            }

        }
    }
}

File: Examples/SampleApp/SampleReportApp/Form1.vb
Public Class Form1

    Private file As String

    Private Sub ButtonSelectReport_Click(sender As Object, e As EventArgs) Handles ButtonSelectReport.Click

        If OpenFileDialog1.ShowDialog = System.Windows.Forms.DialogResult.OK Then

            file = OpenFileDialog1.FileName

        End If

    End Sub

    Private Async Sub ButtonReloadReport_Click(sender As Object, e As EventArgs) Handles ButtonReloadReport.Click
        Await RdlViewer1.SetSourceFile(New Uri(file))
        RdlViewer1.Parameters = "ConnectionString=" & TextBox1.Text
        Await RdlViewer1.Rebuild()
    End Sub


    Private Sub RdlViewer1_PageNavigation(sender As Object, e As fyiReporting.RdlViewer.PageNavigationEventArgs) Handles RdlViewer1.PageNavigation
        MessageBox.Show(e.NewPage)
    End Sub
End Class
